art/utils.py

```diff
@@ -630,11 +638,17 @@ def compute_success_array(
     :param batch_size: Batch size.
     :return: Percentage of successful adversarial samples.
     """
-    adv_preds = np.argmax(classifier.predict(x_adv, batch_size=batch_size), axis=1)
+    preds = classifier.predict(x_clean, batch_size=batch_size)
```

Comment:
Can we move all lines related to `preds` into the `else` block after line 651 to save model evaluations?